### PR TITLE
Fix Gatewayproxy Controller and Re-Enable APIGW v2 Controller

### PR DESCRIPTION
**Pull request by jm96441n**
### Description
- fixes typo in gateway proxy controller
- re-enables apigw v2 controller
- 
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: internal/mesh/internal/controllers/gatewayproxy/controller.go [hunk 1]
```diff
@@ -75,7 +78,7 @@ func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req c
 		return nil
 	}
 
-	switch workload.Metadata["gateway-type"] {
+	switch workload.Metadata[GatewayKindMetadataKey] {
```
Comment (nathancoleman) [inline internal/mesh/internal/controllers/gatewayproxy/controller.go:81]: ✔️ verified that this typo does not exist in the `release/1.18.x` branch

**Review by nathancoleman — COMMENTED** [review]
LGTM. I think you mentioned writing some test coverage for this guy?

**Review by NiniOak — APPROVED** [review]
Thanks for the quick fix

**Comment by jm96441n** [comment]
yeah @nathancoleman I want to get this in to unblock/unbreak the setup and going to follow up with a PR with some tests

**Pull request merged**